kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,665 @@
+Please make sure server name is correct."
+
+  eval $__isValidServer=false
+  eval $__clusterName=UNKNOWN
+  configMap=$(${kubernetesCli} get cm ${domainUid}-weblogic-domain-introspect-cm \
+    -n ${domainNamespace} -o json)
+  topology=$(echo "${configMap}" | jq '.data["topology.yaml"]')
+  jsonTopology=$(python -c \
+    'import sys, yaml, json; print json.dumps(yaml.safe_load('"${topology}"'), indent=4)')
+  servers=($(echo $jsonTopology | jq -r '.domain.servers[].name'))
+  if  checkStringInArray "${serverName}" "${servers[@]}" ; then
+    eval $__clusterName=""
+    eval $__isValidServer=true
+  else
+    dynamicClause=".domain.configuredClusters[] | select (.dynamicServersConfig != null)"
+    namePrefixSize=". | {name: .name, prefix:.dynamicServersConfig.serverNamePrefix, \
+                 max:.dynamicServersConfig.maxDynamicClusterSize}"
+    dynamicClusters=($(echo $jsonTopology | jq "${dynamicClause}" | jq -cr "${namePrefixSize}"))
+    dynamicClustersSize=${#dynamicClusters[@]}
+    for dynaClusterNamePrefix in ${dynamicClusters[@]:-}; do
+      prefix=$(echo ${dynaClusterNamePrefix} | jq -r .prefix)
+      if [[ "${serverName}" == "${prefix}"* ]]; then
+        maxSize=$(echo ${dynaClusterNamePrefix} | jq -r .max)
+        number='^[0-9]+$'
+        if [ $(echo "${serverName}" | grep -c -Eo '[0-9]+$') -gt 0 ]; then
+          serverCount=$(echo "${serverName}" | grep -Eo '[0-9]+$')
+        fi
+        if ! [[ $serverCount =~ $number ]] ; then
+           echo "error: Server name is not valid for dynamic cluster." 
+           exit 1
+        fi
+        if [ "${serverCount}" -gt "${maxSize}" ]; then
+          printError "${errorMessage}"
+          exit 1
+        fi
+        eval $__clusterName="'$(echo ${dynaClusterNamePrefix} | jq -r .name)'"
+        eval $__isValidServer=true
+        break
+      fi
+    done
+    staticClause=".domain.configuredClusters[] | select (.dynamicServersConfig == null)"
+    nameCmd=" . | {name: .name, serverName: .servers[].name}"
+    configuredClusters=($(echo $jsonTopology | jq "${staticClause}" | jq -cr "${nameCmd}"))
+    configuredClusterSize=${#configuredClusters[@]}
+    for configuredClusterName in ${configuredClusters[@]:-}; do
+      name=$(echo ${configuredClusterName} | jq -r .serverName)
+      if [ "${serverName}" == "${name}" ]; then
+        eval $__clusterName="'$(echo ${configuredClusterName} | jq -r .name)'"
+        eval $__isValidServer=true
+        break
+      fi
+    done
+  fi
+}
+
+#
+# Function to validate whether a cluster is valid and part of the domain
+# $1 - Domain unique id.
+# $2 - Domain namespace.
+# $3 - cluster name
+# $4 - Retrun value "true" or "false" indicating whether cluster name is valid
+#
+function validateClusterName {
+  local domainUid=$1
+  local domainNamespace=$2
+  local clusterName=$3
+  local __isValidCluster=$4
+
+  configMap=$(${kubernetesCli} get cm ${domainUid}-weblogic-domain-introspect-cm \
+    -n ${domainNamespace} -o json)
```

Comment:
Is this another occurrence of `get cm` that should have a helpful message if it fails?